src/com/seafile/seadroid2/cameraupload/StubContentProvider.java

```diff
@@ -0,0 +1,66 @@
+package com.seafile.seadroid2.cameraupload;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.net.Uri;
+
+/**
+ * This stub content provider is just glue code between the Authenticator
+ * and the SyncProvider
+ */
+public class StubContentProvider extends ContentProvider {
```

Comment:
Interesting idea, hadn't thought of that.

However, SeafileProvider has some special configurations in the AndroidManifest (especially enabled="@bool/atLeastKitKat"). Using the SyncAdapter with the SeafileProvider would leave the path the Android tutorial has outlined. It might work, it might not work. Or worse, it might cause subtle bugs. I'd rather not risk that.